File: Round 1A 2013/A.cpp
// Problem: Bullseye
// Language: C++
// Author: KirarinSnow
// Usage: g++ thisfile.cpp -o executable && ./executable <input.in >output.out
// Comments: Does not work for large.


#include <iostream>
#include <string>

using namespace std;


int main()
{
    int cases;
    cin >> cases;

    for (int i = 1; i <= cases; i++)
    {
      int r, t;
      cin >> r >> t;
      
      int c = 0;
      while (t >= 0)
      {
	t -= (r+c+1)*(r+c+1) - (r+c)*(r+c);
	c += 2;
      }
      
      cout << "Case #" << i << ": " << c/2-1 << endl;
    }
    
    return 0;
}

File: Round 1A 2013/A.py
#!/usr/bin/env python
#
# Problem: Bullseye
# Language: Python
# Author: KirarinSnow
# Usage: python thisfile.py <input.in >output.out


from math import *

for case in range(int(raw_input())):
    r, t = map(int, raw_input().split())

    # quadratic equation?? offset by 1000 to handle precision
    c = max(0, int(ceil(sqrt((2*r+3)**2+8*(t-2*r-1))/4.0-(2*r+3)/4.0))-1000)
    
    # brute-force near the target to get the exact answer
    while 2*c*(c+1) + (2*r+1)*(c+1) <= t:
        c += 1    

    ans = c
    print "Case #%d: %d" % (case+1, ans)
